rfcs/convergence/root-as-slot.md

```diff
@@ -0,0 +1,241 @@
+  state: ComponentState<R>,
+  slotNames: (keyof R)[] = [],
+): {
+  slots: Slots<R>;
+  slotProps: SlotProps<R>;
+} {
+  const slots = {} as Slots<R>;
+  const slotProps = {} as SlotProps<R>;
+
+  for (const slotName of slotNames) {
+    const [slot, props] = getSlot(state, slotName);
+    slots[slotName] = slot;
+    slotProps[slotName] = props;
+  }
+  return { slots, slotProps: slotProps };
+}
+```
+
+## Pros and Cons
+
+### Pros
+
+1. Treating `root` as a simple slot simplifies _Typings_
+2. Simplification of _Typings_ also comes with better inference performance
+3. References available on every _slot_
+4. Not using `getNativeElementProps` means possible performance improvement
+
+### Cons
+
+1. [Widening Types problem](#⚠️-Widening-types-problem-⚠️)
+2. Explicit of _slots_ declaration makes easier for developers to make mistakes
+
+[1st rule of aria opt-out mechanism]: https://github.com/microsoft/fluentui/blob/master/rfcs/convergence/first-rule-of-aria.md#detailed-proposal
+[do not support as prop for slots that render native dom elements]: https://github.com/microsoft/fluentui/blob/master/rfcs/convergence/simplify-prop-merging.md#do-not-support-as-prop-for-slots-that-render-native-dom-elements
+[primary slot]: https://github.com/microsoft/fluentui/blob/91e25b8615a4a0ddb52b18ba5a51ffe21711bb7c/rfcs/convergence/native-element-props.md
+
+#### ⚠️ [Widening types](https://www.typescriptlang.org/play#example/type-widening-and-narrowing) problem ⚠️
+
+Since _Typings_ for `root` is a subset of the interface that declares the properties of a given component, widening mechanism from assigning types will ensure that `root` _Typings_ are compatible with props which is not necessarily true! This implicates in some conflicts on properties spreading through state and `root` slot.
+
+The problem can be verified in the next example:
+
+```tsx
+type ComponentProps<S extends ObjectShorthandPropsRecord> = Omit<
+  { [Key in keyof S]?: ShorthandProps<S[Key]> },
+  'root'
+> &
+  S['root']; // this will make Slots['root'] part of Props
+
+type Slots = {
+  root: ObjectShorthandProps<React.HTMLAttributes<HTMLElement>, HTMLElement>;
+};
+
+interface Props extends ComponentProps<Slots> {
+  /**
+   * Property that can be passed through props
+   * but doesn't belong on the element itself,
+   * it's only used internally by state
+   */
+  stateSpecificProperty?: unknown;
+}
+interface State extends ComponentState<Slots> {
+  stateSpecificProperty: unknown;
+}
+
+// ...
+// use*State hook
+return {
+  // This should error 🚨 because of stateSpecificProperty,
+  // but it doesn't due to type widening
+  root: resolveShorthand(props),
+};
+```
+
+A easy solution would be to use `getNativeElementProps` to filter out properties in this specific case:
+
+```tsx
+// use*State hook
+return {
+  // This filters out stateSpecificProperty ✅
+  root: resolveShorthand(getNativeElementProps(state.components.root, props)),
```

Comment:
might be interesting to state smwhere that this specific case, can be tested by conformance tests...it could test that custom props aren't leaking into the root DOM element

Reply:
Nice, I'll add that session! and IMO that's the way to go with this, instead of relying on prop filtering we should ensure this by conformance and linting (perhaps)